src/genie/main_genie_elt.py

```diff
@@ -0,0 +1,229 @@
+            release_file_ent.path, sep="\t", comment="#", low_memory=False
+        )
+        if df.empty:
+            logger.warning(f"Skipping empty clinical file: {release_file_key}")
+            continue
+
+        write_to_snowflake(
+            conn=conn,
+            table_df=df,
+            table_name=table_name,
+            overwrite=overwrite,
+            write_pandas_kwargs={"quote_identifiers":False},
+        )
+
+
+def is_valid_release_path(release_path: str) -> bool:
+    """Checks that the current release path
+    is valid.
+
+    - Expects form of  {top_level_release_folder}/{release_name}
+        as the release path, e.g: "Releases/Release 0"
+    - Expects a newer release not present in RELEASES_TO_SKIP
+
+    Args:
+        release_path (str): the input release path to check for
+            validity
+
+    Returns:
+        bool: whether the release path is valid to iterate through
+            or not
+    """
+    release_path_list = release_path[0].split("/")
+    if len(release_path_list) != 2:
+        return False
+    # skips older releases
+    elif release_path_list[1] in RELEASES_TO_SKIP:
+        return False
+    else:
+        return True
+
+
+def main(
+    database: str,
+    overwrite: bool,
+    conn: "snowflake.connector.SnowflakeConnection" = None,
+):
+    """
+    Main function -loops through the
+        top level releases project folder, checks for
+        valid release paths and runs the function to
+        ingest those data into snowflake
+
+    Args:
+        overwrite (bool): Whether to overwrite table data or not
+        database (str): Database table to run ELT commands in
+        conn (snowflake.connector.SnowflakeConnection): Optional Snowflake
+            connection injected externally (e.g. Airflow)
+    """
+    syn = synapseclient.login()
+    conn_obj = get_connection(conn=conn)
+
+    try:
+        logger.info("Connected to Snowflake.")
+        for dirpath, dirnames, _ in synu.walk(syn, PROJECT_SYNID):
+            if is_valid_release_path(release_path=dirpath):
+                for dirname, dir_synid in dirnames:
+                    logger.info(f"Processing release: {dirname}")
+                    push_cbio_files_to_snowflake(
+                        syn=syn,
+                        conn=conn_obj,
+                        synid=dir_synid,
+                        overwrite=overwrite,
+                        database=database,
+                    )
+            else:
+                logger.info(f"Skipping invalid release path: {dirpath}.")
+            logger.info("ELT completed successfully.")
+    finally:
+        # if it's a local connection, close it
+        if conn is None:
+            conn_obj.close()
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Run the Main GENIE ELT pipeline.")
+    parser.add_argument(
+        "--overwrite", action="store_true", help="Overwrite table data if present."
+    )
+    parser.add_argument(
```

Comment:
TODO: Have a restriction on which database(s) these scripts can create the necessary schemas and tables in depending on discussion in https://github.com/Sage-Bionetworks/snowflake/pull/268